framework/src/constraints/AutomaticMortarGeneration.C

```diff
@@ -1850,10 +1850,13 @@ AutomaticMortarGeneration::projectSecondaryNodesSinglePair(
                                                                          secondary_val);
                   }
 
-              // Sanity check
               if (!found_match)
-                mooseError("Could not associate primary/secondary neighbors on either side of "
-                           "secondary_node.");
+              {
+                // There could be coincident nodes and this might be a bad primary candidate (see
+                // issue #21680). Instead of giving up, let's try continuing
+                rejected_primary_elem_candidates.insert(primary_elem_candidate);
+                continue;
```

Comment:
Is there any use in having a `mooseDoOnce(mooseWarning)` here informing on possible coincident nodes?
It may be a relatively infrequent situation.

Reply:
I think it would make sense to do a `mooseDoOnce(mooseWarning)` if we had a demonstrated case in which continuing here was a bad thing. But to my knowledge, well really only based on the fact that nothing bad happened in our test suite, there doesn't appear to be any downside to continuing as opposed to erroring. If there is no apparent downside, I don't want to warn the user for doing something that seems normal to them, e.g. having coincident nodes which seems to be a desired use case for @hugary1995

Reply:
Yeah, that makes sense. My concern here is that it's hard (for me) to predict all possible cases and, in this case, the change is made having a particular situation in mind. A `mooseInfo` would work just as good in my mind. But I am good without any output.